### PR TITLE
control agent updates in e2e tests and fix uts

### PR DESCRIPTION
**Pull request by nagworld9**
## Description

Added new flag to control agent updates in e2e tests. And also, this PR included some unit tests fix which are failed after rebase with develop.

Issue # 

---

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/Azure/WALinuxAgent/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: tests/ga/test_agent_update.py [hunk 1]
```diff
@@ -117,7 +118,7 @@ def test_it_should_not_agent_update_if_last_attempted_update_time_not_elapsed(se
         with self.__get_agent_update_handler(test_data=data_file, autoupdate_frequency=10) as (agent_update_handler, mock_telemetry):
             agent_update_handler._protocol.mock_wire_data.set_extension_config_requested_version(version)
             agent_update_handler._protocol.mock_wire_data.set_incarnation(2)
-            agent_update_handler._protocol.update_goal_state()
+            agent_update_handler._protocol.client.update_goal_state()
```
Comment (nagworld9) [inline tests/ga/test_agent_update.py:121]: This method removed in develop recently as part of protocol changes.

File: azurelinuxagent/common/conf.py [hunk 2]
```diff
@@ -634,6 +634,14 @@ def get_enable_ga_versioning(conf=__conf__):
     return conf.get_switch("Debug.EnableGAVersioning", True)
 
 
+def get_enable_agent_update_in_dcr(conf=__conf__):
+    """
+    If True, the agent checks for updating the agent.
```
Comment (narrieta) [inline azurelinuxagent/common/conf.py:639]: can you clarify this comment? not sure it describes the purpose of the flag

File: azurelinuxagent/ga/agent_update.py [hunk 3]
```diff
@@ -182,8 +182,8 @@ def __log_event(level, msg_, success_=True):
 
     def run(self, goal_state):
         try:
-            # Ignore new agents if update is disabled
-            if not self._autoupdate_enabled:
+            # Ignore new agents if update is disabled. The DCR flag will be removed after testing
```
Comment (narrieta) [inline azurelinuxagent/ga/agent_update.py:185]: do you mean you will remove the config flag once you are done with your tests?

File: azurelinuxagent/ga/agent_update.py [hunk 4]
```diff
@@ -182,8 +182,8 @@ def __log_event(level, msg_, success_=True):
...
+            if not self._autoupdate_enabled or not conf.get_enable_agent_update_in_dcr():
```
Comment (narrieta) [inline azurelinuxagent/ga/agent_update.py:186]: can we improve the name of the flag? (assuming it is not just a temporary flag)

**Review by narrieta — APPROVED** [review]

**Review by maddieford — APPROVED** [review]

File: azurelinuxagent/common/conf.py (hunk 2 above)
Comment (narrieta) [reply azurelinuxagent/common/conf.py:639]: (assuming it is not just a temporary flag)

Comment (narrieta) [reply azurelinuxagent/common/conf.py:639]: if it is temporary, adding a comment REMOVE BEFORE MERGE TO DEVELOP or something like that may be a good idea

**Comment by nagworld9** [comment]
@narrieta Actually this flag is needed all the time to control agent updates in e2e tests either now or future. I have updated according to that. By default, is true. Regarding e2e tests, when we install test version, we set to false (stop agent updates) so that all the test suits run on test version. If particular test wants differently, they need to have extra setup to enable the flag. 

Let me know if you have questions or concerns.

**Comment by narrieta** [comment]
@nagworld9 thanks for the updates. This matches my understanding. It was the previous comment abut deleting the flag after testing that really threw me off.

**Review by narrieta — APPROVED** [review]

**Comment by nagworld9** [comment]
> @nagworld9 thanks for the updates. This matches my understanding. It was the previous comment about deleting the flag after testing that really threw me off.

Sorry for the confusion. I initially thought for doing it temporarily for feature branch and later come up with real flag when we merge to develop. 

I'm glad that we are on same page now. Thanks for the review.

**Pull request merged**